throttle.go

```diff
@@ -0,0 +1,55 @@
+package taskgroup
+
+// A Throttle rate-limits the number of concurrent goroutines that can execute
+// in parallel to some fixed number.  A zero Throttle is ready for use, but
+// imposes no limit on parallel execution. See [Throttle.Enter] for use.
+type Throttle struct {
+	adm chan struct{}
+}
+
+// NewThrottle constructs a [Throttle] with a capacity of n goroutines.
+// If n ≤ 0, the resulting Throttle imposes no limit.
+func NewThrottle(n int) Throttle {
+	if n <= 0 {
+		return Throttle{}
+	}
+	return Throttle{adm: make(chan struct{}, n)}
```

Comment:
Extremely mild thought: for large limiters, a mutex+cond+uint may be more efficient? More irritating to program, but it's all hidden away neatly, so maybe?...

Reply:
I did some testing on it some while back (before I started using this pattern), and it's actually not—I think the compiler may have some special cases for channel-of-zero-width-values or something.